packages/ai-ide/src/common/architect-prompt-template.ts

```diff
@@ -44,3 +46,26 @@ Always look at the relevant files to understand your task using the function ~{$
 {{prompt:project-info}}
 `
 };
+
+export const architectTaskSummaryPromptTemplate: PromptTemplate = {
+    id: ARCHITECT_TASK_SUMMARY_PROMPT_TEMPLATE_ID,
+    template: `{{!-- This prompt is licensed under the MIT License (https://opensource.org/license/mit).
+Made improvements or adaptations to this prompt template? We'd love for you to share it with the community! Contribute back here:
+https://github.com/eclipse-theia/theia/discussions/new?category=prompt-template-contribution --}}
+
+Your task is to analyze the current chat session and summarize it to prepare completing the coding task.
+Your instructions should be complete, they are used by a coding agent.
+Include all necessary information. 
+Use unique identifiers such as file paths or URIs to artifacts.
+Skip irrelevant information, e.g. for discussions, only sum up the final result.
+
+## Instructions
+1. Analyze the conversation carefully.
+2. Identify the main coding objective and requirements.
+3. Propose a clear approach to implement the requested functionality in task steps.
+4. Ask clarifying questions if any part of the task is ambiguous.
```

Comment:
I'm not sure about this part. As the system currently works, we're going to treat the output as the summary, and the agent won't have an opportunity to actually pose questions.

Maybe something like this?

```suggestion
4. If any part of the task is ambiguous, note the ambiguity so that it can be clarified later.
```